File: identity-server/src/main/java/org/identity/identityserver/model/response/ApplicationResponse.java
package org.identity.identityserver.model.response;

import lombok.Getter;
import lombok.Setter;
import lombok.experimental.Accessors;
import org.identity.identityserver.model.enumable.Status;
import org.identity.identityserver.model.response.base.CommonResponse;

import java.util.Set;

/**
 * @author BUI_QUANG_HIEU
 * 5/24/2025
 * ApplicationResponse.java
 */
@Getter
@Setter
@Accessors(chain = true)
public class ApplicationResponse extends CommonResponse {
    private String name;
    private String description;
    private String clientId;
    private String clientSecret;
    private Set<String> redirectUris;
    private String scope;
    private Boolean googleLoginEnable;
    private Set<String> webOrigins;
    private Status status;
}

File: identity-client/src/model/response/ApplicationResponse.ts
import { Status } from '@/model/enumable/Status.ts'
import { ResponseCommon } from '@/model/Common.ts'

export class ApplicationResponse extends ResponseCommon{
  name: string = ''
  clientId: string = ''
  clientSecret: string = ''
  redirectUris: string[] = []
  webOrigins: string[] = []
  description: string = ''
  status: Status = Status.IN_ACTIVE
  statusCheckbox: boolean = false

  constructor(data?: Partial<ApplicationResponse>) {
    super()
    if (data) {
      Object.assign(this, data)
      this.redirectUris = data.redirectUris ? [...data.redirectUris] : []
      this.webOrigins = data.webOrigins ? [...data.webOrigins] : []
      this.statusCheckbox = this.status === Status.ACTIVE
    }
  }
}
